libs/@blockprotocol/type-system/crate/src/ontology/url/mod.rs

```diff
@@ -115,38 +114,18 @@ impl FromStr for VersionedUrl {
     type Err = ParseVersionedUrlError;
 
     fn from_str(url: &str) -> Result<Self, ParseVersionedUrlError> {
-        static RE: LazyLock<Regex> =
-            LazyLock::new(|| Regex::new(r#"(.+/)v/(\d+)(.*)"#).expect("regex failed to compile"));
-
-        if url.len() > 2048 {
-            return Err(ParseVersionedUrlError::TooLong);
-        }
-
-        let captures = RE
-            .captures(url)
-            .ok_or(ParseVersionedUrlError::IncorrectFormatting)?;
-        let base_url = captures
-            .get(1)
-            .ok_or(ParseVersionedUrlError::MissingBaseUrl)?
-            .as_str();
-        let version = captures
-            .get(2)
-            .ok_or(ParseVersionedUrlError::MissingVersion)?
-            .as_str();
-
-        if let Some(suffix) = captures.get(3) {
-            // Regex returns an empty string for capturing groups that don't match anything
-            if !suffix.as_str().is_empty() {
-                return Err(ParseVersionedUrlError::AdditionalEndContent);
-            }
-        }
-
-        Ok(Self {
-            base_url: BaseUrl::new(base_url.to_owned())
-                .map_err(ParseVersionedUrlError::InvalidBaseUrl)?,
-            version: u32::from_str(version)
-                .map_err(|error| ParseVersionedUrlError::InvalidVersion(error.to_string()))?,
-        })
+        url.rsplit_once("v/").map_or(
+            Err(ParseVersionedUrlError::MissingVersion),
+            |(base_url, version)| {
+                Ok(Self {
+                    base_url: BaseUrl::new(base_url.to_owned())
+                        .map_err(ParseVersionedUrlError::InvalidBaseUrl)?,
+                    version: version.parse::<u32>().map_err(|error| {
```

Comment:
If this is an empty string I think we should return `MissingVersion`

Reply:
Also ideally if there is a number, and then more stuff (for example query params) we would warn about `AdditionalEndContent` still

Reply:
Supporting `AdditionalEndContent` is possible but hard as we would search for any non-digit character on our own. Having additional content would result in an error like
> invalid version `12-a-b-c`: invalid digit found in string

Reply:
> is possible but hard as we would search for any non-digit character on our own. Having additional content would result in an error like

It shouldn't be too hard right? **Once the version parse fails** Can't we just iterate over the characters after the `v/` separator and return the slice after the first non-digit?

Reply:
If you pass in `/v/123,456` (accidentally used `,` as thousand separators), you would get `,456` as additional content. Unless you have an overflow error in parsing it would always result in `AdditionalEndContent`. I don't see a reason why we need to catch this, this is basically the same as the `InvalidDigit` error kind returned from parsing.

Reply:
I personally think this is actually quite helpful, it identifies `,456` as the problematic bit, and especially if there's a query param (which I think is far more likely than a comma) you will see something like ``"Additional end content `?q=abstract%20api`"``

Reply:
> If this is an empty string I think we should return MissingVersion

Should be fixed